cognite/client/data_classes/vision.py

```diff
@@ -0,0 +1,255 @@
+        # Replace the loaded VisionExtractPredictions with its corresponding dict representation
+        if "predictions" in item_dump and isinstance(self._predictions_dict, Dict):
+            item_dump["predictions"] = (
+                self._predictions_dict if camel_case else resource_to_snake_case(self._predictions_dict)
+            )
+        return item_dump
+
+    @staticmethod
+    def _process_predictions_dict(predictions_dict: Dict[str, Any]) -> VisionExtractPredictions:
+        """Converts a (validated) predictions dict to a corresponding VisionExtractPredictions"""
+        prediction_object = VisionExtractPredictions()
+        snake_case_predictions_dict = resource_to_snake_case(predictions_dict)
+        for key, value in snake_case_predictions_dict.items():
+            if hasattr(prediction_object, key):
+                feature_class = VISION_FEATURE_MAP[key]
+                setattr(prediction_object, key, [feature_class(**v) for v in value])
+        return prediction_object
+
+
+class VisionExtractJob(VisionJob):
+    _JOB_TYPE = ContextualizationJobType.VISION
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._items: Optional[List[VisionExtractItem]] = None
+
+    def __getitem__(self, file_id: InternalId) -> VisionExtractItem:
+        """Retrieves the results for a file by id"""
+        found = [item for item in self.result["items"] if item.get("fileId") == file_id]
+        if not found:
+            raise IndexError(f"File with id {file_id} not found in results")
+        return VisionExtractItem._load(found[0], cognite_client=self._cognite_client)
+
+    @property
+    def items(self) -> Optional[List[VisionExtractItem]]:
+        """Returns a list of all predictions by file"""
+        if self.status == JobStatus.COMPLETED.value:
+            self._items = [
+                VisionExtractItem._load(item, cognite_client=self._cognite_client) for item in self.result["items"]
+            ]
+        return self._items
+
+    @items.setter
+    def items(self, items: List[VisionExtractItem]) -> None:
+        self._items = items
+
+    @property
+    def errors(self) -> List[str]:
+        """Returns a list of all error messages across files"""
+        return [item["errorMessage"] for item in self.result["items"] if "errorMessage" in item]
+
+    def _predictions_to_annotations(
+        self,
+        creating_user: Optional[str] = None,
+        creating_app: Optional[str] = None,
+        creating_app_version: Optional[str] = None,
+    ) -> List[Annotation]:
+
+        return [
+            Annotation(
+                annotated_resource_id=item.file_id,
+                annotation_type=VISION_ANNOTATION_TYPE_MAP[prediction_type],
+                data=data.dump(),
+                annotated_resource_type="file",
+                status="suggested",
+                creating_app=creating_app or "cognite-sdk",
```

Comment:
```suggestion
                creating_app=creating_app or "cognite-sdk-python",
```

Or is this creating_app too specific, perhaps?

Reply:
[f9d74e2](https://github.com/cognitedata/cognite-sdk-python/pull/1014/commits/f9d74e24d6cf6718fcda63e4b1b355831b038417)